pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -0,0 +1,87 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+// +kubebuilder:object:generate=true
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetType describes the subnet type.
+type SubnetType string
+
+const (
+	SubnetTypePublic  SubnetType = "public"
+	SubnetTypePrivate SubnetType = "private"
+)
+
+// VPCNetworkConfigurationSpec defines the desired state of VPCNetworkConfiguration.
+// There is a default VPCNetworkConfiguration that applies to Namespaces
+// do not have a VPCNetworkConfiguration assigned. When a field is not set
+// in a Namespace's VPCNetworkConfiguration, the Namespace will use the value
+// in the default VPCNetworkConfiguration.
+type VPCNetworkConfigurationSpec struct {
+	// Namespaces to be applied.
+	// The default VPCNetworkConfiguration will not have AppliedToNamespaces set.
+	AppliedToNamespaces []string `json:"appliedToNamespaces,omitempty"`
+	// PolicyPath of Tier0 or Tier0 VRF gateway.
+	DefaultGatewayPath string `json:"defaultGatewayPath,omitempty"`
+	// Edge cluster path on which the networking elements will be created.
+	EdgeClusterPath string `json:"edgeClusterPath,omitempty"`
+	// NSX-T Project the Namespace associated with.
+	NSXTProject string `json:"nsxtProject,omitempty"`
+	// NSX-T IPv4 Block paths used to allocate public Subnets.
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=5
+	PublicIPv4Blocks []string `json:"publicIPv4Blocks,omitempty"`
+	// Private IPv4 CIDRs used to allocate private Subnets.
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=5
+	PrivateIPv4CIDRs []string `json:"privateIPv4CIDRs,omitempty"`
+	// Default subnet prefix length for Subnet.
+	// Defaults to 26.
+	// +kubebuilder:default=26
+	DefaultSubnetPrefixLength int `json:"defaultSubnetPrefixLength,omitempty"`
```

Comment:
Should we use "DefaultIPv4SubnetSize" if NSX-T API uses SubnetSize (?).

Reply:
changed. thanks Jianjun